src/frontend/src/pages/AdminToolsPage/ProjectsConfig/UsefulLinks/UsefulLinkFormModal.tsx

```diff
@@ -0,0 +1,109 @@
+import { Controller, useForm } from 'react-hook-form';
+import NERFormModal from '../../../../components/NERFormModal';
+import { FormControl, FormLabel, FormHelperText, Box, Grid, Select, MenuItem } from '@mui/material';
+import ReactHookTextField from '../../../../components/ReactHookTextField';
+import { useToast } from '../../../../hooks/toasts.hooks';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { Link, LinkCreateArgs, LinkType } from 'shared';
+import { linkToLinkCreateArgs } from '../../../../utils/link.utils';
+
+interface UsefulLinkFormModalProps {
+  open: boolean;
+  handleClose: () => void;
+  clickedLink?: Link;
```

Comment:
So that you don't have to pull out of an optional to check against things, you could accept default values in this component that are pulled out of the link in the edit component